File: 2025/01/solution1.lua
#!/usr/bin/env lua

local input_file = 'input.txt'
local position = 50
local count = 0

-- Process each rotation
for line in io.lines(input_file) do
  line = line:match("^%s*(.-)%s*$")  -- strip whitespace
  if line == "" then
    goto continue
  end

  -- Parse rotation: L or R followed by a number
  local direction, distance_str = line:match("^([LR])(%d+)$")
  if not direction then
    goto continue
  end

  local distance = tonumber(distance_str)

  -- Apply rotation
  if direction == 'L' then
    position = (position - distance) % 100
  else -- direction == 'R'
    position = (position + distance) % 100
  end

  -- Count if position is 0
  if position == 0 then
    count = count + 1
  end

  ::continue::
end

print(count)

File: 2025/01/solution1.py
#!/usr/bin/env python3

input_file = 'input.txt'
position = 50
count = 0

# Process each rotation
with open(input_file, 'r') as f:
    for line in f:
        line = line.strip()
        if not line:
            continue

        # Parse rotation: L or R followed by a number
        if line[0] not in ['L', 'R']:
            continue
        
        direction = line[0]
        distance = int(line[1:])

        # Apply rotation
        if direction == 'L':
            position = (position - distance) % 100
        else:  # direction == 'R'
            position = (position + distance) % 100

        # Count if position is 0
        if position == 0:
            count += 1

print(count)
